jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipResponseAndCallback.java

```diff
@@ -220,14 +222,23 @@ protected void commit(boolean last, Callback callback, ByteBuffer content)
             return;
         }
 
+        // We are varying the response due to accept encoding header.
+        if (_vary != null)
+            fields.ensureField(_vary);
+
+        // If there is nothing to write, don't compress.
+        if (last && BufferUtil.isEmpty(content))
+        {
+            LOG.debug("{} exclude by nothing to write", this);
```

Comment:
Use `if (LOG.isDebugEnabled())`

Reply:
Ok, but will change the whole class, which for some reason never used this pattern.